--- judgedaemon.main.php-720	2020-03-01 21:00:22.000000000 +0800
+++ judgedaemon.main.php-mod	2020-03-01 21:11:51.693388000 +0800
@@ -106,6 +106,10 @@
 
     curl_setopt($curl_handle, CURLOPT_CUSTOMREQUEST, $verb);
     curl_setopt($curl_handle, CURLOPT_HTTPHEADER, []);
+
+    curl_setopt($curl_handle, CURLOPT_SSL_VERIFYPEER, 0);
+    curl_setopt($curl_handle, CURLOPT_SSL_VERIFYHOST, 0);
+
     if ($verb == 'POST') {
         curl_setopt($curl_handle, CURLOPT_POST, true);
         if (is_array($data)) {
@@ -384,6 +388,7 @@
             logmsg(LOG_DEBUG, "Compiling");
             $olddir = getcwd();
             chdir($execpath);
+            system("chmod +x ./build");
             system("./build >> " . LOGFILE . " 2>&1", $retval);
             if ($retval!=0) {
                 return array(null, "Could not run ./build in $execpath.");
@@ -1050,6 +1055,12 @@
             'metadata' => rest_encode_file($testcasedir . '/program.meta', $output_storage_limit),
             'output_diff'  => rest_encode_file($testcasedir . '/feedback/judgemessage.txt', $output_storage_limit)
         );
+
+        if (isset($row['full_judge']) && $row['full_judge'] == false) {
+            unset($new_judging_run['output_run']);
+            unset($new_judging_run['output_error']);
+        }
+
         $unsent_judging_runs[] = $new_judging_run;
         $outstanding_data += strlen(var_export($new_judging_run, TRUE));
 
